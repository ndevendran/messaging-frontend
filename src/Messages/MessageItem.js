import React from 'react';
import { Router, Link } from 'react-router-dom';

export default ({ message }) => {
  return (
    <div className="message">
      <Link to={`/view/${message.id}`}>
      <div className="header">
        <small>{message.user.username}</small>
      </div>
      <div className="body">
        <p>{message.text}</p>
      </div>
      <div className="footer">
      </div>
      </Link>
    </div>
  );
}
