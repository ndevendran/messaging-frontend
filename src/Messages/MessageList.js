import React from 'react';
import MessageItem from './MessageItem.js';
import { withRouter } from 'react-router-dom';

export default ({ messages }) => {
  return (
    messages.map((message) =>
      <div key={message.id}>
        <MessageItem message={message} />
      </div>
    )
  );
}
