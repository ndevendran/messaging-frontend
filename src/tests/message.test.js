import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MessageList from '../Messages/MessageList.js';
import MessageItem from '../Messages/MessageItem.js';


Enzyme.configure({adapter: new Adapter()});

describe('Message', () => {
  const messages = [
    {
      id: 1,
      user: {
        username: 'testuser1',
      },
      text: 'test message one',
    },
    {
      id: 2,
      user: {
        username: 'testuser2',
      },
      text: 'test message two',
    },
  ];
  describe('Message List', () => {
    it('has a valid snapshot', () => {
      const component = renderer.create(
        <MessageList messages={messages} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<MessageList messages={messages} />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders all messages in the list', () => {
      const element = shallow(
        <MessageList messages={messages} />
      );

      expect(element.find(MessageItem).length).toBe(2);
    });
  });

  describe('Message Item', () => {
    it('has a valid snapshot', () => {
      const component = renderer.create(
        <MessageItem message={messages[0]} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('contains header, footer, and body all under a message class', () => {
      const element = shallow(
        <MessageItem message={messages[1]} />
      );
      expect(element.find('.message').length).toBe(1);
      expect(element.find('.footer').length).toBe(1);
      expect(element.find('.header').length).toBe(1);
      expect(element.find('.body').length).toBe(1);
    });

  });

  describe('Message Footer', () => {
    it('renders without crashing', () => {
      
    });
  });
});
