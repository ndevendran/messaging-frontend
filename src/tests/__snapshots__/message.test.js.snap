// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Message Item has a valid snapshot 1`] = `
<div
  className="message"
>
  <div
    className="header"
  >
    <small>
      testuser1
    </small>
  </div>
  <div
    className="body"
  >
    <p>
      test message one
    </p>
  </div>
  <div
    className="footer"
  />
</div>
`;

exports[`Message Message List has a valid snapshot 1`] = `
Array [
  <div>
    <div
      className="message"
    >
      <div
        className="header"
      >
        <small>
          testuser1
        </small>
      </div>
      <div
        className="body"
      >
        <p>
          test message one
        </p>
      </div>
      <div
        className="footer"
      />
    </div>
  </div>,
  <div>
    <div
      className="message"
    >
      <div
        className="header"
      >
        <small>
          testuser2
        </small>
      </div>
      <div
        className="body"
      >
        <p>
          test message two
        </p>
      </div>
      <div
        className="footer"
      />
    </div>
  </div>,
]
`;
